### DISCUSSION TITLE
Support for overrides yaml for DropwizardTestSupport

### DISCUSSION THREAD

**Discussion by sandeepnkulkarni** [dropwizard/dropwizard#8530] [Development]
We are using DropwizardTestSupport in TestNG. Till date we have not faced any issue with using ConfigOverride while starting up Dropwizard for testing purpose.

But as application has got bigger and complex, we feel a need to provide complete set of overrides via overrides yaml itself.

So my question is, can this be added as a feature in some point in future release?

**Comment by zUniQueX** [comment]
Hi @sandeepnkulkarni. I think supplying this from the framework could be confusing very fast. You then would have to specify a second `ConfigurationSourceProvider` or something similar. I personally don't see the benefit of providing a second YAML with overrides directly alongside the 'normal' application YAML instead of combining them into one YAML file. 

If you have that much `ConfigOverride`s and can't modify the application YAML, you probably could implement this yourselves very similar to what's done applying the `ConfigOverrides` in `dropwizard-configuration`. To use most built-in methods, I think I would implement this parsing the second YAML and generating a set of `ConfigOverride` objects to provide to the `DropwizardAppExtension`. Traversing the second YAML file to get the `ConfigOverride`s could use most of the logic of traversing the original `JsonNode` to [apply a `ConfigOverride`](https://github.com/dropwizard/dropwizard/blob/release/3.0.x/dropwizard-configuration/src/main/java/io/dropwizard/configuration/BaseConfigurationFactory.java#L188-L266).
